mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -849,6 +969,18 @@ LogicalResult tosa::ConcatOp::inferReturnTypeComponents(
   return success();
 }
 
+LogicalResult tosa::ConcatOp::verify() {
+  // check that each input has same element type as output
+  auto outType = getOutput().getType();
```

Comment:
Use llvm::all_of?

Reply:
Yes, we can do that.

Reply:
change refactored to this patch: https://github.com/llvm/llvm-project/pull/127923

Reply:
https://github.com/llvm/llvm-project/pull/127923 merged